src/transformers/models/whisper/generation_whisper.py

```diff
@@ -652,10 +606,17 @@ def generate(
                 batch_idx_map=batch_idx_map,
             )
 
-            # 6.5 prepare decoder input ids
-            suppress_tokens = _get_attr_from_logit_processors(
-                logits_processor, SuppressTokensLogitsProcessor, "suppress_tokens"
-            )
+            # 6.3 prepare decoder input ids
+            if logits_processor is not None: 
+                suppress_tokens = _get_attr_from_logit_processors(
+                    logits_processor, SuppressTokensLogitsProcessor, "suppress_tokens"
+                )
+            else: 
+                suppress_tokens = None
+
+            # Remove decoder_input_ids if present in kwargs:
+            decoder_input_ids = kwargs.pop("decoder_input_ids", None)
```

Comment:
So we pop them here and then don't do anything with them? Is this intended? Shouldn't we pop them from the kwargs, and if they're None, then initialise them from the generation config? 

This logic should go in `_prepare_decoder_input_ids` either way